extensions/blocks/contact-info/contact-info.php

```diff
@@ -10,31 +10,135 @@
+	)
 );
 
 /**
- * Contact info block registration/dependency declaration.
+ * Class Jetpack_Contact_Info_Block
  *
- * @param array  $attr    Array containing the contact info block attributes.
- * @param string $content String containing the contact info block content.
+ * Helper class that lets us add schema attributes dynamically because they are not something that is store with the content
  *
- * @return string
  */
-function jetpack_contact_info_block_load_assets( $attr, $content ) {
-	Jetpack_Gutenberg::load_assets_as_required( 'contact-info' );
-	return $content;
+class Jetpack_Contact_Info_Block {
+
+	/**
+	 * Adds contact info schema attributes
+	 *
+	 * @param array  $attr    Array containing the contact info block attributes.
+	 * @param string $content String containing the contact info block content.
+	 *
+	 * @return string
+	 */
+	static function render( $attr, $content ) {
+		Jetpack_Gutenberg::load_assets_as_required( 'contact-info' );
+
+		return str_replace(
+			'class="wp-block-jetpack-contact-info"',
+			'class="wp-block-jetpack-contact-info" itemprop="location" itemscope itemtype="http://schema.org/Organization"',
+			$content );
+	}
+
+	/**
+	 * Adds address schema attributes
+	 *
+	 * @param array  $attr    Array containing the address block attributes.
+	 * @param string $content String containing the address block content.
+	 *
+	 * @return string
+	 */
+	static function render_address( $attr, $content ) {
+		// return empty content if the only attribute set is linkToGoogleMaps
+		if ( ! self::has_attributes( $attr, array( 'linkToGoogleMaps' ) ) ) {
+			return '';
+		}
+		$find = array(
+			'class="wp-block-jetpack-address"',
+			'class="jetpack-address__address', // Closing " left out on purpose - there are multiple address fields and they all need to be updated with the same itemprop
+			'class="jetpack-address__region"',
+			'class="jetpack-address__city"',
+			'class="jetpack-address__postal"',
+			'class="jetpack-address__country"',
+		);
+		$replace = array(
+			'itemprop="address" itemscope itemtype="http://schema.org/PostalAddress" class="wp-block-jetpack-address" ',
+			'itemprop="streetAddress" class="jetpack-address__address', // Closing " left out on purpose
+			'itemprop="addressRegion" class="jetpack-address__region"',
+			'itemprop="addressLocality" class="jetpack-address__city"',
+			'itemprop="postalCode" class="jetpack-address__postal"',
+			'itemprop="addressCountry" class="jetpack-address__country"',
+		);
+		return str_replace( $find, $replace, $content );
+	}
+
+	/**
+	 * Adds email schema attributes
+	 *
+	 * @param array  $attr    Array containing the email block attributes.
+	 * @param string $content String containing the email block content.
+	 *
+	 * @return string
+	 */
+	static function render_email( $attr, $content ) {
+		if ( ! self::has_attributes( $attr ) ) {
+			return ''; // nothing to see here
+		}
+		return str_replace( 'href="mailto:', 'itemprop="email" href="mailto:', $content );
```

Comment:
Don't think it's necessary to negate the `self::has_attributes` in these functions. Also, consider using a ternary for brevity:
```php
return self::has_attributes( $attr )
    ? str_replace( 'href="mailto:', 'itemprop="email" href="mailto:', $content )
    : '';
```

Reply:
great suggestion. Thanks